src/Data/Enum.purs

```diff
@@ -211,13 +215,13 @@ instance boundedEnumEither :: (BoundedEnum a, BoundedEnum b) => BoundedEnum (Eit
       | n >= ca && n < cab = Right <$> toEnum (n - ca)
       | otherwise = Nothing
   fromEnum (Left a) = fromEnum a
-  fromEnum (Right b) = fromEnum b + runCardinality (cardinality :: Cardinality a)
+  fromEnum (Right b) = fromEnum b + unwrap (cardinality :: Cardinality a)
 
 instance boundedEnumTuple :: (BoundedEnum a, BoundedEnum b) => BoundedEnum (Tuple a b) where
   cardinality =
     Cardinality
-      $ runCardinality (cardinality :: Cardinality a)
-      * runCardinality (cardinality :: Cardinality b)
+      $ unwrap (cardinality :: Cardinality a)
```

Comment:
Oh right, I did try it in fact, but I remember now: it doesn't help with almost any of these cases since we're combining `Cardinality` values with different phantom types